tendermint/starport
# feat(network): campaign update

close #2070
close #2071
close #2085

## Description

- Implement campaign update command to update the campaign name, total shares, and total supply;
- Add `total-supply`, `total-shares` and `metadata` to update the total campaign shares to the publish command;

### How to test

- Update the campaign
```shell
starport n --local campaign update 1 --name foo --total-supply 10token --total-shares 10uspn --from alice
```

- Or run the publish with the `total-supply` flag
```shell
starport n --local chain publish https://github.com/lubtd/earth --total-supply 100uspn --from alice
```

- Verify the new infos
```shell
spnd query campaign show-campaign 1
```
